src/Marten/Events/CodeGeneration/AggregateEventProcessingFrame.cs

```diff
@@ -78,10 +78,8 @@ public override void GenerateCode(GeneratedMethod method, ISourceWriter writer)
 
             if (Apply != null)
             {
-                if (AggregateType.GetConstructors().Any(x => !x.GetParameters().Any()))
-                {
-                    writer.Write($"{Aggregate.Usage} ??= new {AggregateType.FullNameInCode()}();");
-                }
+                var awaitStatement = IsAsync ? "await " : " ";
+                writer.Write($"{Aggregate.Usage} ??= {awaitStatement}Create({SpecificEvent.Usage}, session);");
```

Comment:
So at this spot, *if it exists*, you'd delegate to `CreationFrame.GenerateCode()` here, but first you'd set that up to assign the new aggregate to the `Aggregate` variable

Reply:
Few lines above it's used like that: https://github.com/JasperFx/marten/blob/fixe_inline_projection_for_type_with_a_private_constructor/src/Marten/Events/CodeGeneration/AggregateEventProcessingFrame.cs#L77.

However, it's not assigned with the reasoning I explained:  https://github.com/JasperFx/marten/pull/1770#issuecomment-798964344 and https://github.com/JasperFx/marten/pull/1770#issuecomment-798965494.